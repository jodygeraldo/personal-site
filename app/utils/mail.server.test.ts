import { sendMail, validateMailRequest } from './mail.server'

let badMail = {
  name: 'jody geraldo',
  email: 'jody.geraldo.jg@gmail.com',
  message: 'abcde',
}

let goodMail = {
  name: 'jody geraldo',
  email: 'jody.geraldo.jg@gmail.com',
  message:
    'Lorem ipsum dolor sit amet consectetur adipisicing elit. Laudantium, voluptatibus.',
}

beforeEach(() => {
  badMail = {
    name: 'jody geraldo',
    email: 'jody.geraldo.jg@gmail.com',
    message: 'abcde',
  }

  goodMail = {
    name: 'jody geraldo',
    email: 'jody.geraldo.jg@gmail.com',
    message:
      'Lorem ipsum dolor sit amet consectetur adipisicing elit. Laudantium, voluptatibus.',
  }
})

test('validateMailRequest should return fieldErrors and formError for bad form request', () => {
  expect(validateMailRequest(badMail)).toMatchObject({
    fieldErrors: {
      message: 'Message is required and must be at least 20 characters',
    },
    formError: 'There were errors with your submission',
  })

  expect(validateMailRequest(badMail)).toMatchSnapshot()

  badMail.name = 'ab'

  expect(validateMailRequest(badMail)).toMatchSnapshot()

  badMail.email = ''
  expect(validateMailRequest(badMail)).toMatchSnapshot()
})

test('validateMailRequest should return empty object fieldErrors and undefined formError', () => {
  expect(validateMailRequest(goodMail)).toMatchObject({
    fieldErrors: {},
    formError: undefined,
  })
})

test('sendMail should return error object if message contains bad words', async () => {
  goodMail.message =
    'this message is not a good message and contains a bad word like ASS and bitch or cuck'

  expect(await sendMail('API_KEY', goodMail)).toMatchSnapshot()
})

test.todo('sendMail should return success object on success')

test.todo('sendMail should return error object if send mail fail')
