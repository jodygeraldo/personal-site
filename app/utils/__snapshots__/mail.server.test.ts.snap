// Vitest Snapshot v1

exports[`sendMail should return error object if message contains bad words 1`] = `
{
  "extendedMessage": "If you believe this is a mistake, please contact me through other means",
  "message": "The message you sent, contains blacklisted words",
  "type": "ERROR",
}
`;

exports[`validateMailRequest should return fieldErrors and formError for bad form request 1`] = `
{
  "fieldErrors": {
    "message": "Message is required and must be at least 20 characters",
  },
  "formError": "There were errors with your submission",
}
`;

exports[`validateMailRequest should return fieldErrors and formError for bad form request 2`] = `
{
  "fieldErrors": {
    "message": "Message is required and must be at least 20 characters",
    "name": "Name is required and must be at least 3 characters",
  },
  "formError": "There were errors with your submission",
}
`;

exports[`validateMailRequest should return fieldErrors and formError for bad form request 3`] = `
{
  "fieldErrors": {
    "email": "Email is required",
    "message": "Message is required and must be at least 20 characters",
    "name": "Name is required and must be at least 3 characters",
  },
  "formError": "There were errors with your submission",
}
`;
